src/lib/comms/sol-lwm2m.c

```diff
@@ -2198,13 +2225,84 @@ sol_lwm2m_resource_init(struct sol_lwm2m_resource *resource,
         uint16_t until = i;
 
         for (i = 0; i < until; i++)
-            sol_blob_unref(resource->data[i].blob);
+            sol_blob_unref(resource->data[i].content.blob);
     }
     free(resource->data);
     va_end(ap);
     return r;
 }
 
+SOL_API int
+sol_lwm2m_resource_init_vector(struct sol_lwm2m_resource *resource,
+    uint16_t id, enum sol_lwm2m_resource_data_type data_type,
+    struct sol_vector *res_instances)
+{
+    uint16_t i;
+    int r = -EINVAL;
+    uint16_t resource_len = res_instances->len;
+
+    if (!resource || data_type == SOL_LWM2M_RESOURCE_DATA_TYPE_NONE ||
+        !resource_len)
+        return -EINVAL;
```

Comment:
Not really, I just based myself on existing code: https://github.com/solettaproject/soletta/pull/2264/files#diff-e4058dc14a343a8aceb7210b323358a2R2169

I will change it in both places.